Libraries/LibELF/Validation.cpp

```diff
@@ -205,6 +205,12 @@ bool validate_program_headers(const Elf32_Ehdr& elf_header, size_t file_size, co
                 *interpreter_path = String((const char*)&buffer[program_header.p_offset], program_header.p_filesz - 1);
             break;
         case PT_LOAD:
+            if (program_header.p_filesz > program_header.p_memsz) {
+                if (verbose)
+                    dbgln("Program header ({}) PT_LOAD has p_filesz ({}) larger than p_memsz ({})", header_index, program_header.p_filesz, program_header.p_memsz);
+                return false;
+            }
+            break;
```

Comment:
This change makes PT_LOAD headers skip the validation of p_offset and p_flags that used to be common to LOAD, DYNAMIC, NOTE, PHDR, and TLS. Should either muse `[[ fallthrough ]]` or consider if the validation you're should apply to more than just PT_LOAD

Reply:
> This change makes PT_LOAD headers skip the validation of p_offset and p_flags that used to be common to LOAD

How does it cause validation to be skipped?

This check is in the `validate_program_headers()` function. If this check fails it will return false. Presumably any consuming function should bail out if the header validation fails.

> or consider if the validation you're adding should apply to more than just PT_LOAD

I tried apply this check to more than `PT_LOAD` (all program headers) but ran into an issue with `PT_NOTE` in coredumps, as coredumps have a hardcoded `memsz` of zero.

Also, applying this check only to `PT_LOAD` is inline with the Linux [implementation](https://elixir.bootlin.com/linux/latest/source/fs/binfmt_elf.c#L663) and [documentation](https://man7.org/linux/man-pages/man5/elf.5.html).

Reply:
The break statement. Before, the checks on line 218 through 227 were called for PT_LOAD. But with the new break here on line 213, if a PT_LOAD succeeds your new check, the old p_offset and p_flags checks are ignored for that program header.

Reply:
ugh thanks. I didn't realize that was intentional.

kling has merged my PR to set `memsz` to a non-zero value for coredumps, so I'll revert my changes back to what I had originally. That is, performing this check for every program header.